StRoot/StRefMultCorr/StBadRunChecker.cxx

```diff
@@ -0,0 +1,317 @@
+    mEnergy=RunEnergy;
+    mColMode=CollisionMode;
+    mSpecies=Species;
+    TString name_tmp=mRun+"_"+mColMode+"_"+mEnergy+"_"+mSpecies;
+    mRunIndex=-1;
+    for(int i=0;i<mRunNameCount;i++){
+        if(name_tmp==mRunNameList[i]){
+            mRunIndex=i;
+            break;
+        }
+    }
+    readBadRunsFromHeaderFile();
+}
+
+//______________________________________________________________________________
+// Default destructor
+StBadRunChecker::~StBadRunChecker() {
+  /* empty */
+}
+
+//______________________________________________________________________________
+void StBadRunChecker::readBadRunsFromHeaderFile() {
+    switch(mRunIndex){
+        case 0:
+            for(int i=0;i<2;i++){
+                mRunRange.push_back(run19_col_19p6_auau_range[i]);
+            }
+            for(int i=0;i<nBadRun_run19_col_19p6_auau;i++){
+                mBadRun_all.push_back(badrun_run19_col_19p6_auau_all[i]);
+                std::vector<Int_t> row;
+                for(int j=0;j<12;j++){
+                    row.push_back(badrun_run19_col_19p6_auau_sub[i][j]);       
+                }
+                mBadRun_sub.push_back(row);
+            } 
+            cout<<"read in bad run list for "<<mRun<<" "<<mColMode<<" "<<mSpecies<<" "<<mEnergy<<" GeV. "<<nBadRun_run19_col_19p6_auau<<" bad runs in total."<<endl;
+            break;
+        /*
+        case 1:
+            for(int i=0;i<2;i++){
+                mRunRange.push_back(run19_col_14p6_auau_range[i]);
+            }
+            for(int i=0;i<nBadRun_run19_col_14p6_auau;i++){
+                mBadRun_all.push_back(badrun_run19_col_14p6_auau_all[i]);
+                std::vector<Int_t> row;
+                for(int j=0;j<12;j++){
+                    row.push_back(badrun_run19_col_14p6_auau_sub[i][j]);       
+                }
+                mBadRun_sub.push_back(row);
+            } 
+            cout<<"read in bad run list for "<<mRun<<" "<<mColMode<<" "<<mSpecies<<" "<<mEnergy<<" GeV. "<<nBadRun_run19_col_14p6_auau<<" bad runs in total."<<endl;
+            break;
+        */
+        //Add more runs here
+        default:
+            cout<<mRun<<" "<<mColMode<<" "<<mSpecies<<" "<<mEnergy<<" GeV doesn't exist or the bad run list is not available yet.Exit!"<<endl;
+            exit(0);
+    }
+
+}
+
+
+Bool_t StBadRunChecker::isBadRun(const Int_t RunId,TString mS){
+    // Return true if a given run id is bad run
+    if (RunId<mRunRange[0]||RunId>mRunRange[1]){
+        cout<<"Warning: "<<RunId<<" is not in "<<mRun<<" "<<mColMode<<" "<<mSpecies<<" "<<mEnergy<<"GeV. Exit!"<<endl;
+        exit(0);
+    }
+    mS.ToLower(); 
+    vector<Int_t>::iterator iter = std::find(mBadRun_all.begin(), mBadRun_all.end(), RunId);
+    if(iter == mBadRun_all.end()){
+        return 0;//RunId is not a bad run 
+    }
+    //return ( iter != mBadRun.end() ) ;
+    else{
+        int index = std::distance(mBadRun_all.begin(), iter);//index starts at zero
+        int isbad=1;
```

Comment:
If I understand correctly, you use "isbad=0" as the run being bad and "isbad=1" being good, which is a bit counter-intuitive. This is also opposite to the convention used in "badrun_*_sub".

Reply:
You are right. I want to implement a "or" logic in this method, so it is easier to set isbad=0 for bad runs. I can change the name to "isgood" to avoid confusion.